common/djangoapps/embargo/api.py

```diff
@@ -0,0 +1,242 @@
+    if ip_addr in IPFilter.current().blacklist_ips:
+        return True
+        # return REASONS['ip_blacklist'].format(
+        # ip_addr=ip_addr,
+        # from_course=_from_course_msg(course_id, course_is_embargoed)
+        # )
+
+    # If we're white-listed, then allow access
+    if ip_addr in IPFilter.current().whitelist_ips:
+        return None
+
+
+def _check_user_country(country_code, course_id=u""):
+    # Retrieve the country code from the IP address
+    # and check it against the list of embargoed countries
+    return CountryAccessRule.check_country_access(course_id, country_code)
+
+
+def get_user_country_from_profile(user, course_id=""):
+    """
+    Check whether the user is embargoed based on the country code in the user's profile.
+
+    Args:
+        user (User): The user attempting to access courseware.
+
+    Keyword Args:
+        course_id (unicode): The course the user is trying to access.
+        course_is_embargoed (boolean): Whether the course the user is accessing has been embargoed.
+
+    Returns:
+        A unicode message if the user is embargoed, otherwise `None`
+
+    """
+    cache_key = u'user.{user_id}.profile.country'.format(user_id=user.id)
+    profile_country = cache.get(cache_key)
+    if profile_country is None:
+        profile = getattr(user, 'profile', None)
+        if profile is not None and profile.country.code is not None:
+            profile_country = profile.country.code.upper()
+        else:
+            profile_country = ""
+        cache.set(cache_key, profile_country)
+
+    return profile_country
+
+def _country_code_from_ip(ip_addr):
+    """
+    Return the country code associated with an IP address.
+    Handles both IPv4 and IPv6 addresses.
+
+    Args:
+        ip_addr (str): The IP address to look up.
+
+    Returns:
+        str: A 2-letter country code.
+
+    """
+    if ip_addr.find(':') >= 0:
+        return pygeoip.GeoIP(settings.GEOIPV6_PATH).country_code_by_addr(ip_addr)
+    else:
+        return pygeoip.GeoIP(settings.GEOIP_PATH).country_code_by_addr(ip_addr)
+
+
+def check_access(user, ip_address, course_key):
+    """
+    Check is the user with this ip_address has access to the given course
+
+    Params:
+        user (User): Currently logged in user object
+        ip_address (str): The ip_address of user
+        course_key (CourseLocator): CourseLocator object the user is trying to access
+
+    Returns:
+        Redirect to a URL configured in Django settings or a forbidden response if any constraints fails or None
+
+    """
+
+    course_is_restricted = RestrictedCourse.is_restricted_course(course_key)
+    # If they're trying to access a course that cares about embargoes
+
+    if not course_is_restricted:
+        return None
+
+    if _check_ip_lists(ip_address, course_key):
+        return REASONS['ip_blacklist'].format(
+            ip_addr=ip_address,
+            from_course=_from_course_msg(course_key, course_is_restricted)
+        )
```

Comment:
I think this can just return True or False.  We can add in tracking log events indicating the reason later, but I don't think we need to return this information to the caller.